Storage/SQL/SQLiteHistoryFavicons.swift

```diff
@@ -0,0 +1,377 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Alamofire
+import Fuzi
+import SDWebImage
+import Shared
+import XCGLogger
+
+private let log = Logger.syncLogger
+
+// Set up Alamofire for downloading web content for parsing.
+// NOTE: We use the desktop UA to try and get hi-res icons.
+private var alamofire: SessionManager = {
+    var sessionManager: SessionManager!
+    DispatchQueue.main.sync {
+        let configuration = URLSessionConfiguration.default
+        var defaultHeaders = SessionManager.default.session.configuration.httpAdditionalHeaders ?? [:]
+        defaultHeaders["User-Agent"] = UserAgent.desktopUserAgent()
+        configuration.httpAdditionalHeaders = defaultHeaders
+        configuration.timeoutIntervalForRequest = 5
+        sessionManager = SessionManager(configuration: configuration)
+    }
+    return sessionManager
+}()
+
+// If all else fails, this is the default "default" icon.
+private var defaultFavicon: UIImage = {
+    return UIImage(named: "defaultFavicon")!
+}()
+
+// An in-memory cache of "default" favicons keyed by the
+// first character of a site's domain name.
+private var defaultFaviconImageCache = [String : UIImage]()
+
+class FaviconLookupError: MaybeErrorType {
+    let siteURL: String
+    init(siteURL: String) {
+        self.siteURL = siteURL
+    }
+    var description: String {
+        return "Unable to find favicon for site URL: \(siteURL)"
+    }
+}
+
+class FaviconDownloadError: MaybeErrorType {
+    let faviconURL: String
+    init(faviconURL: String) {
+        self.faviconURL = faviconURL
+    }
+    internal var description: String {
+        return "Unable to download favicon at URL: \(faviconURL)"
+    }
+}
+
+extension SQLiteHistory: Favicons {
```

Comment:
Was this accidental? This file shouldn't be added here otherwise we're gonna collide when we merge the rust-bookmarks branch.